### ISSUE TITLE
Frontend dialog boxes don't appear reliably in iOS browsers or webviews

### ISSUE THREAD

**Issue by rccoleman** [hacs/integration#1267]
## Installation details

| Description                | Value |
| -------------------------- | ----- |
| HACS version               | 1.03
| Home Assistant version     | 0.110.3
| Installation method for HA | Supervised via KVM on Ubuntu
| Web browser                | iOS webview
| Web browser version        | iOS 13.5

## Checklist

- [x] I'm running the newest version of HACS https://github.com/hacs/integration/releases/latest
- [x] I have filled out the issue template to the best of my ability.
- [x] I have read https://hacs.xyz/docs/issues
- [x] This issue is related to the frontend of HACS.
- [x] This issue only contain 1 issue (if you have multiple issues, open one issue for each issue).

## Describe the issue

When using the HACS frontend via an iOS browser or webview in the Home Assistant iOS app, taps often don't respond the first time (need a second tap) or at all.  Sometimes popup dialogs don't appear, either.  It just feels like the interface isn't very responsive compared to the rest of Home Assistant in the same browser.

### Steps to reproduce

One good example is the "reinstall" button on an existing integration.  On a PC web browser, I get a popup that lets me select a version to install.  On iOS, nothing appears and I stay on the screen with the installed integrations listed.

Another is that the "About HACS" item in the main three-dot menu at the top right also doesn't display a popup on iOS, but works fine in a desktop browser.

### Screenshots

This is the reinstall button that does nothing in iOS, and works properly on a desktop web browser:

![IMG_1027](https://user-images.githubusercontent.com/860888/82961023-03acd280-9f71-11ea-9608-3c38fe51e3a1.jpeg)

</details>

**Comment by ludeeus** [comment]
Clear your browser cache.
It's using an older version of the frontend.

**Comment by rccoleman** [comment]
Hmm. I had cleared the frontend csche via the option in the iOS app settings and it reset my login/theme/etc. without fixing this. I went ahead and cleared everything from the phone settings (history, cache, cookies, etc), force quit the app, force quit the browser, and I still see the same behavior as I described in both places.

**Comment by rccoleman** [comment]
If you still think this is a cache issue, I'm not sure what more I can do on my phone.  Someone else on Discord mentioned something similar: https://discordapp.com/channels/330944238910963714/551871772484698112/715291890873925733

**Comment by raetha** [comment]
I'll add a me too to this. Done the front end cache clearing as well, and many functions in the new HACS UI so not work from Safari or the HA app on my iPhone running iOS 13.5 and HA 110.4. They generally work from desktop, but anything that seems to pop up an overlay simply will not display or work on mobile. Tempted to roll back HACS until fixed.

Let me know if I can help provide any testing or detailed information.

**Comment by rccoleman** [comment]
This is still happening with HACS 1.1.0. I’ve tried all manner of resetting/emptying the cache, from the Reset Frontend Cache button in the app to clearing all history and data from settings and the same issue persists. I have to reselecr my theme when I do that, so I’m pretty sure that the browser is actually clearing data.

I have several Lovelace entities installed that I don’t use, which HACS now alerts me to on the main page. Unfortunately, I cannot uninstall them because the ‘uninstall’ menu option just does nothing. I assume that it’s trying to display a pop up and is hitting the original issue described here.

I also periodically see the icon for HACS missing from the sidebar, only displaying the HACS text. This only happens on iOS and seeks to come and go. It’s happening now, even after all attempts to clear the cache and reload.

![FF881410-06DF-45D3-A717-DCCAAB949AF4](https://user-images.githubusercontent.com/860888/83969853-24392e80-a887-11ea-8033-faae0b6464f2.png)

**Comment by ludeeus** [comment]
You need to get the browser logs.
I can not reproduce this with safari.

**Comment by rccoleman** [comment]
I get no logs from a chrome://inspect running in another tab while navigating HACS, and I only see the normal logs when I go back to the overview. This is from starting in HACS and triggering the issue several times (try to uninstall items or pull down the About HACS option in the 3-dot menu), going back to the overview page, back into HACS and trying to the trigger the issue again.

```
http://hassio.socalrob.com:8123/lovelace/0
INFO%cFOLD-ENTITY-ROW 1.0.2 IS INSTALLED color: green; font-weight: bold 
INFO%cCARD-TOOLS 2.1.2 IS INSTALLED
  %cDeviceID: e26e01ce-0059a13a color: green; font-weight: bold 
INFO%cCARD-MOD 1.1.0 IS INSTALLED color: green; font-weight: bold 
INFO%c MINI-GRAPH-CARD %c 0.9.3  color: white; background: coral; font-weight: 700; color: coral; background: white; font-weight: 700;
INFO%c  CUSTOM-HEADER  
%c  Version 1.5.2   color: orange; font-weight: bold; background: black color: white; font-weight: bold; background: dimgray
LOGPreloaded card auto-entities
LOGPreloaded card weather-card
LOGPreloaded card mini-graph-card
LOGPreloaded card fold-entity-row
LOGPreloaded card upcoming-media-card
http://hassio.socalrob.com:8123/lovelace/0
INFO%cCARD-TOOLS 2.1.2 IS INSTALLED
  %cDeviceID: e26e01ce-0059a13a color: green; font-weight: bold 
INFO%cFOLD-ENTITY-ROW 1.0.2 IS INSTALLED color: green; font-weight: bold 
INFO%cCARD-MOD 1.1.0 IS INSTALLED color: green; font-weight: bold 
INFO%c MINI-GRAPH-CARD %c 0.9.3  color: white; background: coral; font-weight: 700; color: coral; background: white; font-weight: 700;
LOGPreloaded card auto-entities
LOGPreloaded card weather-card
LOGPreloaded card mini-graph-card
LOGPreloaded card fold-entity-row
LOGPreloaded card upcoming-media-card
INFO%c  CUSTOM-HEADER  
%c  Version 1.5.2   color: orange; font-weight: bold; background: black color: white; font-weight: bold; background: dimgray
```

**Comment by FeijooFdz** [comment]
I have the same problem from an iPad Pro with the latest versions of Hacs.

**Comment by rccoleman** [comment]
Since I'm not the only one having this issue, I'd like to help get to the bottom of it.  Even when I connect my phone to my Macbook and enable the remote web inspector, I don't see any logs at all while I'm navigating HACS from Safari.  This is the entirety of the console logging starting from a reload of my main Lovelace view, navigating to HACS, going into Integrations, pulling down the three-dot menu, and trying to open the "About HACS" menu item:

```
[Info] BROWSER_MOD 1.0.4 IS INSTALLED (browser_mod.js, line 60)
    DeviceID: bd033fa2-7bf85187
[Info] FOLD-ENTITY-ROW 1.0.2 IS INSTALLED – "" (fold-entity-row.js, line 46)
[Info]  MINI-GRAPH-CARD  0.9.3  (mini-graph-card-bundle.js, line 1)
[Info] CARD-TOOLS 2.1.2 IS INSTALLED (card-tools.js, line 1)
  DeviceID: bd033fa2-7bf85187
[Info] CARD-MOD 1.1.0 IS INSTALLED – "" (card-mod.js, line 1)
[Info]   BUTTON-CARD   (button-card.js, line 1597)
 Version 3.3.6 
[Info] LAYOUT-CARD 1.2.1 IS INSTALLED – "" (layout-card.js, line 68)
[Info]   CUSTOM-HEADER   (custom-header.js, line 595)
  Version 1.5.8  
[Log] Preloaded card auto-entities (lovelace-card-preloader.js, line 1)
[Log] Preloaded card weather-card (lovelace-card-preloader.js, line 1)
[Log] Preloaded card mini-graph-card (lovelace-card-preloader.js, line 1)
[Log] Preloaded card fold-entity-row (lovelace-card-preloader.js, line 1)
[Log] Preloaded card upcoming-media-card (lovelace-card-preloader.js, line 1)
[Log] Preloaded card layout-card (lovelace-card-preloader.js, line 1)
[Error] Unhandled Promise Rejection: TypeError: Object(a.c)()._addCard is not a function. (In 'Object(a.c)()._addCard()', 'Object(a.c)()._addCard' is undefined)
	(anonymous function) (card-tools.js:1:14103)
	Promise
	(anonymous function) (card-tools.js:1:13804)
	r (card-tools.js:1:115)
	(anonymous function) (card-tools.js:1:904)
	Module Code (card-tools.js:1:912)
	evaluate
	moduleEvaluation
[Error] Unhandled Promise Rejection: TypeError: null is not an object (evaluating 'e.shadowRoot.appendChild')
	(anonymous function) (fold-entity-row.js:1:6233)
	asyncFunctionResume
	(anonymous function) (fold-entity-row.js:1:6008)
	map
	setConfig (fold-entity-row.js:1:5896)
	s (chunk.6dbb65acb727c33dd251.js:2144:318405)
	value (chunk.6dbb65acb727c33dd251.js:1257:194400)
	value
	(anonymous function) (card-mod.js:1:5796)
	map
	value (chunk.6dbb65acb727c33dd251.js:1191:1046)
	update (app.cb642fbe.js:2:20366)
	performUpdate (app.cb642fbe.js:2:15560)
	(anonymous function) (app.cb642fbe.js:2:15255)
	asyncFunctionResume
	(anonymous function)
	promiseReactionJobWithoutPromise
[Error] Unhandled Promise Rejection: TypeError: null is not an object (evaluating 'e.shadowRoot.appendChild')
	(anonymous function) (fold-entity-row.js:1:6233)
	asyncFunctionResume
	(anonymous function) (fold-entity-row.js:1:6008)
	map
	setConfig (fold-entity-row.js:1:5896)
	s (chunk.6dbb65acb727c33dd251.js:2144:318405)
	value (chunk.6dbb65acb727c33dd251.js:1257:194400)
	value
	(anonymous function) (card-mod.js:1:5796)
	map
	value (chunk.6dbb65acb727c33dd251.js:1191:1046)
	update (app.cb642fbe.js:2:20366)
	performUpdate (app.cb642fbe.js:2:15560)
	(anonymous function) (app.cb642fbe.js:2:15255)
	asyncFunctionResume
	(anonymous function)
	promiseReactionJobWithoutPromise
	promiseReactionJob
[Error] Failed to load resource: the server responded with a status of 404 () (hls.min.js.map, line 0)
[Log] Current User's Name:  – "Robert Coleman" (custom-header.js, line 595)
[Log] Current User's ID:  – "e7dc6f01443f4e7cbad2acb2929ca81e" (custom-header.js, line 595)
```

There are a few errors in there, but only from the Overview and not from HACS.  This is with popups enabled and all popup blockers disabled, and after hitting "Clear History and Website Data".

@ludeeus What can I do to help make HACS functional on my iPhone?

**Comment by rccoleman** [comment]
This is not fixed in 1.1.1, even after clearing the cache and ensuring that the correct Frontend is being used. Can you please reopen?

**Comment by ludeeus** [comment]
Correct, this will be in 1.1.2 or 1.2.0 the fix was merged after 1.1.1 was released.

**Comment by rccoleman** [comment]
Got it, thanks!
